### ISSUE TITLE
Error: packing sprite failed

### ISSUE THREAD

**Issue by paulocoutinhox** [houmain/spright#14]
Hi,

Im starting get this error:

```
spright -p out
packing sprite failed (65x) in line 13
```

Can you help me?

**Comment by paulocoutinhox** [comment]
Appear be an error here:
https://github.com/houmain/spright/blob/7f2a70e48f849937620b12c5cb16d4df4c40df73/src/packing.cpp#L265

**Comment by houmain** [comment]
Hi, this message means that the sprites did not fit on the sheet.
Does your configuration still look like this?
```
max-width 1024
max-height 1024
output "tank-texture.png"
```
You could either increase the `max-width` or `max-height` or pack the sprites on multiple sheets by providing a sequence instead of a single filename like: `output "tank-texture{0-}.png"`.
Thanks for reporting, I will make the message more helpful.

**Comment by paulocoutinhox** [comment]
Hi, what solve was `output "tank-texture{0-}.png"`, perfect. Thanks again.

**Comment by paulocoutinhox** [comment]
Hi,

Reopening this, because when i set for multiple textures, the plist don't load the textures and don't make the rect in correct PNG file, since now i have some:

<img width="229" alt="image" src="https://github.com/user-attachments/assets/326eccae-3e7b-416a-9463-a15d525bf7b5">


```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>metadata</key>
	<dict>
		<key>format</key>
		<integer>3</integer>
		<key>textureFileName</key>
		<string>effect-texture0.png</string>
		<key>size</key>
		<string>{ 1024, 1024 }</string>
	</dict>
	<key>frames</key>
	<dict>
		<key>bubble</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 1, 1 }</string>
            <key>spriteSize</key>
            <string>{ 126, 126 }</string>
            <key>spriteSourceSize</key>
            <string>{ 126, 126 }</string>
            <key>textureRect</key>
            <string>{ { 131, 890 }, { 126, 126 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_1_1</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 60, 60 }</string>
            <key>spriteSourceSize</key>
            <string>{ 60, 60 }</string>
            <key>textureRect</key>
            <string>{ { 758, 890 }, { 60, 60 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_1_2</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 57, 56 }</string>
            <key>spriteSourceSize</key>
            <string>{ 57, 56 }</string>
            <key>textureRect</key>
            <string>{ { 879, 890 }, { 57, 56 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_1_3</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 64, 63 }</string>
            <key>spriteSourceSize</key>
            <string>{ 64, 63 }</string>
            <key>textureRect</key>
            <string>{ { 565, 890 }, { 64, 63 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_1_4</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 46, 45 }</string>
            <key>spriteSourceSize</key>
            <string>{ 46, 45 }</string>
            <key>textureRect</key>
            <string>{ { 696, 952 }, { 46, 45 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_1_5</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 53, 52 }</string>
            <key>spriteSourceSize</key>
            <string>{ 53, 52 }</string>
            <key>textureRect</key>
            <string>{ { 938, 890 }, { 53, 52 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_2_1</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 41, 49 }</string>
            <key>spriteSourceSize</key>
            <string>{ 41, 49 }</string>
            <key>textureRect</key>
            <string>{ { 820, 948 }, { 41, 49 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_2_2</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 75, 77 }</string>
            <key>spriteSourceSize</key>
            <string>{ 75, 77 }</string>
            <key>textureRect</key>
            <string>{ { 488, 890 }, { 75, 77 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_2_3</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 127, 127 }</string>
            <key>spriteSourceSize</key>
            <string>{ 127, 127 }</string>
            <key>textureRect</key>
            <string>{ { 2, 890 }, { 127, 127 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_crash_1_18</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 16, 3 }</string>
            <key>spriteSize</key>
            <string>{ 494, 442 }</string>
            <key>spriteSourceSize</key>
            <string>{ 494, 442 }</string>
            <key>textureRect</key>
            <string>{ { 2, 2 }, { 494, 442 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_crash_1_19</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 16, 3 }</string>
            <key>spriteSize</key>
            <string>{ 494, 442 }</string>
            <key>spriteSourceSize</key>
            <string>{ 494, 442 }</string>
            <key>textureRect</key>
            <string>{ { 498, 446 }, { 494, 442 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_crash_1_20</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 16, 3 }</string>
            <key>spriteSize</key>
            <string>{ 494, 442 }</string>
            <key>spriteSourceSize</key>
            <string>{ 494, 442 }</string>
            <key>textureRect</key>
            <string>{ { 2, 446 }, { 494, 442 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_crash_1_20</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 16, 3 }</string>
            <key>spriteSize</key>
            <string>{ 494, 442 }</string>
            <key>spriteSourceSize</key>
            <string>{ 494, 442 }</string>
            <key>textureRect</key>
            <string>{ { 498, 2 }, { 494, 442 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_smoke_1_1</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 60, 60 }</string>
            <key>spriteSourceSize</key>
            <string>{ 60, 60 }</string>
            <key>textureRect</key>
            <string>{ { 696, 890 }, { 60, 60 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_smoke_1_2</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 57, 56 }</string>
            <key>spriteSourceSize</key>
            <string>{ 57, 56 }</string>
            <key>textureRect</key>
            <string>{ { 820, 890 }, { 57, 56 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_smoke_1_3</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 63, 63 }</string>
            <key>spriteSourceSize</key>
            <string>{ 63, 63 }</string>
            <key>textureRect</key>
            <string>{ { 631, 890 }, { 63, 63 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_smoke_1_4</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 46, 45 }</string>
            <key>spriteSourceSize</key>
            <string>{ 46, 45 }</string>
            <key>textureRect</key>
            <string>{ { 863, 948 }, { 46, 45 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>explosion_smoke_1_5</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 53, 52 }</string>
            <key>spriteSourceSize</key>
            <string>{ 53, 52 }</string>
            <key>textureRect</key>
            <string>{ { 938, 944 }, { 53, 52 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>smoke_puff_1_1</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 98, 90 }</string>
            <key>spriteSourceSize</key>
            <string>{ 98, 90 }</string>
            <key>textureRect</key>
            <string>{ { 388, 890 }, { 98, 90 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
		<key>smoke_puff_1_2</key>
		<dict>
            <key>spriteOffset</key>
            <string>{ 0, 0 }</string>
            <key>spriteSize</key>
            <string>{ 127, 120 }</string>
            <key>spriteSourceSize</key>
            <string>{ 127, 120 }</string>
            <key>textureRect</key>
            <string>{ { 259, 890 }, { 127, 120 } }</string>
            <key>textureRotated</key>
            <false/>			
            <key>aliases</key>
            <array/>
		</dict>
	</dict>
    <key>aliases</key>
    <array/>
</dict>
</plist>
```

**Comment by houmain** [comment]
Do .plists support more than one texture?
I just tried to pack some sprites using https://free-tex-packer.com/app/ and reduced the width/height so it had to arrange them on multiple textures. When exporting it to cocos2d it created a .zip contianing one .plist file per texture.

Is that what you were expecting? This strategy is unfortunately not available yet in spright.

**Comment by paulocoutinhox** [comment]
This is a good question, and it don't have support.

I think that the best option will be multiple `.plist`, as you sad and work from free-tex-packer.

**Comment by houmain** [comment]
Hi, 
outputting multiple .plist files at once is available in version 3.5.1.
The only thing you have to do is to provide a sequence in `description` like:
```
description "out/tank-texture-{0-}.plist"
```
Then one .plist file is written per slice.

**Comment by paulocoutinhox** [comment]
Thanks, it works.
